Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,245 @@
+            {
+                var symbol = data.Symbol;
+
+                var holdingsTurnover = 1m;
+                var holdingsValue = security.Holdings.AbsoluteHoldingsValue;
+
+                if (holdingsValue != 0)
+                {
+                    // Gets the order of the event to get its value in account currency
+                    var order = _portfolio.Transactions.GetOrderById(orderId);
+                    var orderValue = Math.Abs(order.GetValue(security));
+                    holdingsTurnover = orderValue / holdingsValue;
+
+                    if (holdingsTurnover == 0)
+                    {
+                        RemoveConsolidator(consolidator, symbol);
+                        return;
+                    }
+                }
+
+                var totalMarketDollarVolume = GetTotalMarketDollarVolume(data) * security.SymbolProperties.ContractMultiplier;
+                var totalTradeVolumeCapacity = totalMarketDollarVolume / holdingsTurnover;
+
+                // If it is a new day, we discard the previous day data
+                var utcDate = data.EndTime.ConvertToUtc(security.Exchange.TimeZone).Date;
+                if (_previousInputTime != utcDate)
+                {
+                    _previousInputTime = utcDate;
+                    _maximumSymbolCapacity.Clear();
+                    _historicalPortfolioCapacity.Add(0m);
+                }
+
+                // Updates AUM Capacity if there is new information or lower total trade volume capacity for the security
+                decimal capacity;
+                if (!_maximumSymbolCapacity.TryGetValue(symbol, out capacity) ||
+                    totalTradeVolumeCapacity < capacity)
+                {
+                    _maximumSymbolCapacity[symbol] = totalTradeVolumeCapacity;
+                    _historicalPortfolioCapacity[0] = _maximumSymbolCapacity.Sum(x => x.Value);
+                }
+
+                RemoveConsolidator(consolidator, symbol);
+            }
+        }
+
+        /// <summary>
+        /// Gets the total market volume which is a fraction of the current trade bar volume
+        /// </summary>
+        /// <param name="data">Last price trade bar available or consolidated</param>
+        private static decimal GetTotalMarketDollarVolume(IBaseData data)
+        {
+            var tradeBar = (TradeBar) data;
+
+            // Maximum percentage of market volume that is tradeable
+            var factor = tradeBar.Period == Time.OneDay
+                ? .025m
+                : .050m;
+
+            return factor * tradeBar.Volume * tradeBar.Price;
+        }
+
+        /// <summary>
+        /// Remove consolidator for a given symbol
+        /// </summary>
+        /// <param name="consolidator">Consolidator to be removed</param>
+        /// <param name="symbol">Symbol associated with the consolidator</param>
+        private void RemoveConsolidator(IDataConsolidator consolidator, Symbol symbol)
```

Comment:
This might make a nice extension method -- I can see this being applicable in many places. I like it!